controllers/grid/navigationMenus/form/NavigationMenuForm.inc.php

```diff
@@ -0,0 +1,203 @@
+				->toArray();
+		$unassignedItems = array_udiff($navigationMenuItems, $assignedItems, function($a, $b) {
+			return $a->getId() - $b->getId();
+		});
+
+		if ($this->_navigationMenuId) {
+			$navigationMenuDao = DAORegistry::getDAO('NavigationMenuDAO');
+			$navigationMenu = $navigationMenuDao->getById($this->_navigationMenuId);
+
+			$templateMgr->assign('navigationMenuIsDefault', $navigationMenu->getDefault());
+		} else {
+			$templateMgr->assign('navigationMenuIsDefault', 0);
+		}
+
+
+		$templateMgr->assign(array(
+			'enabledThemes' => $enabledThemes,
+			'activeThemeNavigationAreas' => $activeThemeNavigationAreas,
+			'unassignedItems' => $unassignedItems,
+			'navigationMenuId' => $this->_navigationMenuId,
+			'title' => $this->getData('title'),
+			'navigationMenuArea' => $this->getData('area_name'),
+			'menuTree' => $this->getData('menuTree'),
+		));
+
+		return parent::fetch($request);
+	}
+
+	/**
+	 * Initialize form data from current NavigationMenu.
+	 */
+	function initData() {
+
+		if (isset($this->_navigationMenuId) && $this->_navigationMenuId != 0) {
+			$navigationMenusDao = DAORegistry::getDAO('NavigationMenuDAO');
+			$navigationMenu = $navigationMenusDao->getById($this->_navigationMenuId);
+
+			if ($navigationMenu != null) {
+				$this->_data = array(
+					'title' => $navigationMenu->getTitle(),
+					'navigationMenuId' => $navigationMenu->getId(),
+					'navigationMenuArea' => $navigationMenu->getAreaName(),
+					'menuTree' => $navigationMenu->getMenuTree(),
+				);
+			} else {
+				$this->_navigationMenuId = null;
+			}
+		}
+	}
+
+	/**
+	 * Assign form data to user-submitted data.
+	 */
+	function readInputData() {
+		$this->readUserVars(array('title', 'navigationMenuId', 'area_name', 'menuTree'));
+
+	}
+
+	/**
+	 * Save NavigationMenu .
+	 */
+	function execute() {
+		$navigationMenusDao = DAORegistry::getDAO('NavigationMenuDAO');
+		$navigationMenuItemAssignmentDao = DAORegistry::getDAO('NavigationMenuItemAssignmentDAO');
+
+		if (isset($this->_navigationMenuId)) {
+			$navigationMenu = $navigationMenusDao->getById($this->_navigationMenuId);
+		}
+
+		if (!isset($navigationMenu)) {
+			$navigationMenu = $navigationMenusDao->newDataObject();
+		}
+
+		$navigationMenu->setContextId($this->_contextId);
+		$navigationMenu->setTitle($this->getData('title'));
+		$navigationMenu->setAreaName($this->getData('area_name'));
+
+		// Update or insert NavigationMenu
+		if ($navigationMenu->getId() != null) {
+			$navigationMenusDao->updateObject($navigationMenu);
+		} else {
+			$this->_navigationMenuId = $navigationMenusDao->insertObject($navigationMenu);
+		}
+
+		// Update NavigationMenuItemAssignment
+		if ($this->_navigationMenuId) {
+			$navigationMenuItemAssignmentDao->deleteByMenuId($this->_navigationMenuId);
+			$menuTree = $this->getData('menuTree');
+			if (!empty($menuTree)) {
+				foreach ($menuTree as $menuItemId => $assignmentData) {
+					$assignment = new NavigationMenuItemAssignment();
```

Comment:
Should use the DAO's `newDataObject` method to instantiate data objects rather than doing it directly.